airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/development/DevDatabaseMigrator.java

```diff
@@ -97,6 +107,19 @@ private static DatabaseMigrator getBaselineMigrator(final FlywayDatabaseMigrator
     return new FlywayDatabaseMigrator(fullMigrator.getDatabase(), baselineConfig.load());
   }
 
+  /**
+   * Create a baseline migration from a specified target migration version.
+   */
+  private static DatabaseMigrator getBaselineMigratorForVersion(final FlywayDatabaseMigrator fullMigrator, final MigrationVersion migrationVersion) {
+    final FluentConfiguration baselineConfig = getBaselineConfig(fullMigrator);
+
+    // Set the baseline flyway config to run up to a target migration version.
+    LOGGER.info("Baseline migrator target version: {}", migrationVersion);
+    baselineConfig.target(migrationVersion);
```

Comment:
Note for the future, it feels like we should be using this directly rather than previous logic. In practice, I'd rather have a functions for tests that says apply all migration until X. Where X would be the current migration to test.